internal/kafka/internal/clusters/provider_test.go

```diff
@@ -0,0 +1,120 @@
+package clusters
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/config"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/client/ocm"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/db"
+)
+
+func TestNewDefaultProviderFactory(t *testing.T) {
+	type args struct {
+		ocmClient              ocm.ClusterManagementClient
+		connectionFactory      *db.ConnectionFactory
+		ocmConfig              *ocm.OCMConfig
+		awsConfig              *config.AWSConfig
+		dataplaneClusterConfig *config.DataplaneClusterConfig
+	}
+	tests := []struct {
+		name string
+		args args
+		want *DefaultProviderFactory
+	}{
+		{
+			name: "Should return the default Provider Factory",
+			args: args{},
+			want: &DefaultProviderFactory{
+				providerContainer: map[api.ClusterProviderType]Provider{
+					api.ClusterProviderStandalone: &StandaloneProvider{},
+					api.ClusterProviderOCM:        &OCMProvider{
+						clusterBuilder: &clusterBuilder{
+							idGenerator: ocm.NewIDGenerator("mk-"),
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := NewDefaultProviderFactory(tt.args.ocmClient, tt.args.connectionFactory, tt.args.ocmConfig, tt.args.awsConfig, tt.args.dataplaneClusterConfig); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("NewDefaultProviderFactory() = %v, want %v", got, tt.want)
+			}
```

Comment:
I find this hard to read. I think would be more readable to have it like this:
```suggestion
			got := NewDefaultProviderFactory(tt.args.ocmClient, tt.args.connectionFactory, tt.args.ocmConfig, tt.args.awsConfig, tt.args.dataplaneClusterConfig)
			Expect(got).To(Equal(tt.want))
```